### ISSUE TITLE
Faraday connection can no longer be configured - documentation is incorrect

### ISSUE THREAD

**Issue by jhottenstein** [plaid/plaid-ruby#404]
I think the v14 update broke the faraday tuning example: https://github.com/plaid/plaid-ruby/tree/9f63d69facf8b0c72df7f709ec1692767c0c0236#tuning-faraday

Plaid::Configuration#new does not pass a faraday object to the block and so `options` is undefined for `builder`

**Comment by jhottenstein** [comment]
Context:

We would like to upgrade from v13 to v14 but we have a custom logger that we attach to faraday.  Before we can upgrade, we need to be able to access the faraday connection object.  Looking at the generated code (and template: https://github.com/plaid/plaid-ruby/blob/9f63d69facf8b0c72df7f709ec1692767c0c0236/templates/ruby/api_client_faraday_partial.mustache#L5-L11), it doesn't look like this connection block is exposed anymore.

Should I open a separate issue for this?  Would you accept a pull request?

**Comment by phoenixy1** [comment]
@jhottenstein Thanks for the report! the ruby clib expert is out today, so a more informed answer will have to wait for him to get back, but I'm pretty sure we would accept a PR to fix the template file if you were to submit one.

**Comment by cgfarmer4** [comment]
@jhottenstein can you provide an example of what you've tried and what you're trying to accomplish? Its possible the documentation may be pointing you in the wrong direction. We use the OpenAPI generator for the pillars of this project, we can modify the templates if you have a suggestion on our side to alleviate the concerns.

**Comment by daisy1754** [comment]
@cgfarmer4 I'm facing the similar issue with @jhottenstein 

I want to setup custom faraday middleware for instrumentation. example in README says:

> Tuning Faraday
```
client = Plaid::Configuration.new do |builder|
  Plaid::Configuration.default()

  # Increase network timeout
  builder.options[:timeout] = 60*20    # 20 minutes
end
```

But Faraday connection is not exposed here so we cannot hook anything to underling faraday client

**Comment by cgfarmer4** [comment]
@daisy1754 what are you trying to achieve? Faraday configurations are exposed via the configuration object.

**Comment by brady-kimball** [comment]
@cgfarmer4  while the configuration is exposed, it seems there's no longer the ability to use our own middleware since the builder (Faraday::Connection)  is no longer exposed. Previously we could do something like this (where Plaid::ServerErrorsMiddleware is our own custom middleware, for example): 

```
def build(timeout: nil)
    Plaid::Client.new(env: config[:env], client_id: config[:client_id], secret: config[:secret]) do |builder|
        builder.use Plaid::ServerErrorsMiddleware
    end
end
```

**Comment by cgfarmer4** [comment]
Got it, we'll work on a patch for this. Thanks!

**Comment by alex-mcconnell** [comment]
Hi - I believe I have also ran into this issue attempting to set a timeout via the configuration object. I don't see where the timeout attribute is ever passed to the Faraday connection. I see timeout referenced from the config, but never passed anywhere...

https://github.com/plaid/plaid-ruby/blob/70cae6afcc6c4278784e35f43650a8dcc3653e17/lib/plaid/api_client.rb#L111-L118

**Comment by javierjulio** [comment]
@alex-mcconnell you're right. It sets `req_opts` and does an update but from there it's not used or referenced anywhere. Seeing that made me wonder since upgrading then if our configured timeout is even being applied.

**Comment by vorpus** [comment]
Hi all - please see if `15.4.0` works for your use case.

An accessor for the Farday::Connection object is exposed in ApiClient ([readme](https://github.com/plaid/plaid-ruby#tuning-faraday))

**Comment by javierjulio** [comment]
@vorpus thanks. Worth noting that while example uses `timeout`, the Configuration object has a `timeout` property that is documented but is still not applied due to https://github.com/plaid/plaid-ruby/issues/404#issuecomment-1100316504. That can be reported as a separate issue though. It would help to have that addressed somehow since it is a documented option.

**Comment by daisy1754** [comment]
I realized my request and @brady-kimball 's request is different than original issue, so created a new issue - https://github.com/plaid/plaid-ruby/issues/424

**Comment by jpriollaud** [comment]
@vorpus 14 was broken, 15 worked

**Comment by phoenixy1** [comment]
excellent! closing this one as resolved.
